backend/fms_core/viewsets.py

```diff
@@ -422,8 +443,78 @@ class SampleKindViewSet(viewsets.ModelViewSet):
     queryset = SampleKind.objects.all()
     serializer_class = SampleKindSerializer
     pagination_class = None
-    permission_classes = [AllowAny]
+    permission_classes = [IsAuthenticated]
+
+class ProtocolViewSet(viewsets.ModelViewSet):
+    queryset = Protocol.objects.all()
+    serializer_class = ProtocolSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated]
+
+class ProcessSampleViewSet(viewsets.ModelViewSet, TemplateActionsMixin):
+    queryset = ProcessSample.objects.all().select_related("process").prefetch_related("lineage")
+    queryset = queryset.annotate(child_sample=F("lineage__child"))
+    queryset = queryset.annotate(child_sample_name=F("lineage__child__name"))
+    queryset = queryset.annotate(source_sample_name=F("source_sample__name"))
+
+    serializer_class = ProcessSampleSerializer
+
+    ordering_fields = (
+        *_list_keys(_process_sample_filterset_fields),
```

Comment:
This wont work in the case of CSV export, which uses a CSVRenderer from a library.
In viewset in get_renderer_context, for your action, you will need to declare the header and the labels of your context (see as an example line 558) If you find a better method, feel free to change it, but I had been looking for a while, and havent found anything else.

Reply:
thanks